internal/handlers/pg/msg_findandmodify.go

```diff
@@ -310,6 +344,23 @@ func prepareFindAndModifyParams(document *types.Document) (*findAndModifyParams,
 		return nil, err
 	}
 
+	var maxTimeMS int32
+	if maxTimeMS, err = common.GetOptionalParam(document, "maxTimeMS", maxTimeMS); err != nil {
+		var commonErr *common.Error
+		if errors.As(err, &commonErr) {
+			return nil, getErrorForInvalidMaxTimeMS(document)
+		}
+
+		return nil, err
+	}
+
+	if maxTimeMS < 0 {
+		return nil, common.NewErrorMsg(
+			common.ErrBadValue,
+			fmt.Sprintf("%v value for maxTimeMS is out of range", maxTimeMS),
+		)
+	}
+
```

Comment:
I think getErrorForInvalidMaxTimeMS should be modified to return the correct value too, not only error. That way, we can reduce the amount of copy&pasted code

Reply:
Fixed